src/serialization.hpp

```diff
@@ -124,4 +132,82 @@ Status serializePredictResponse(
     }
     return status;
 }
+template <typename T>
+Status serializePredictResponse(
+    OutputGetter<T>& outputGetter,
+    const tensor_map_t& outputMap,
+    InferenceResponse* response,
+    outputNameChooser_t outputNameChooser) {
+    OVMS_PROFILE_FUNCTION();
+    Status status;
+    ProtoGetter<InferenceResponse*, InferenceTensor&> protoGetter(response);
+    for (const auto& [outputName, outputInfo] : outputMap) {
+        ov::Tensor tensor;
+        status = outputGetter.get(outputNameChooser(outputName, *outputInfo), tensor);
+        if (!status.ok()) {
+            return status;
+        }
+        auto servableMetaPrecision = outputInfo->getPrecision();
+        auto actualPrecision = ovElementTypeToOvmsPrecision(tensor.get_element_type());
+        if (servableMetaPrecision != actualPrecision) {
+            return StatusCode::INTERNAL_ERROR;
+        }
+        if (!outputInfo->getShape().match(tensor.get_shape())) {
+            return StatusCode::INTERNAL_ERROR;
+        }
+        switch (servableMetaPrecision) {
+        case ovms::Precision::FP64:
+        case ovms::Precision::FP32:
+        case ovms::Precision::FP16:
+        case ovms::Precision::I64:
+        case ovms::Precision::I32:
+        case ovms::Precision::I16:
+        case ovms::Precision::I8:
+        case ovms::Precision::U64:
+        case ovms::Precision::U32:
+        case ovms::Precision::U16:
+        case ovms::Precision::U8:
+            break;
+        case ovms::Precision::BF16:
+        case ovms::Precision::U4:
+        case ovms::Precision::U1:
+        case ovms::Precision::BOOL:  // ?
+        case ovms::Precision::CUSTOM:
+        case ovms::Precision::UNDEFINED:
+        case ovms::Precision::DYNAMIC:
+        case ovms::Precision::MIXED:
+        case ovms::Precision::Q78:
+        case ovms::Precision::BIN:
+        default: {
+            Status status = StatusCode::OV_UNSUPPORTED_SERIALIZATION_PRECISION;
+            SPDLOG_ERROR(status.string());
+            return status;
+        }
+        }
+        InferenceTensor* outputTensor{nullptr};
+        status = response->addOutput(
+            outputInfo->getMappedName(),                             // name
```

Comment:
This should rely on either it is DAG/single model case correct? If so leave comment

Reply:
It relies on a fact that name is mapped - for single model: mapping_config.json, for DAG it is pipeline output name setting. Added comment